From d189394216d39332ab06c346e071e3cde62cbf04 Mon Sep 17 00:00:00 2001
From: Roy Pledge <roy.pledge@nxp.com>
Date: Thu, 22 Aug 2019 11:04:25 +0800
Subject: [PATCH 563/741] soc/fsl/qbman: add cache flush for PPC

commit 7ce1aeec996c1a013d7270ad68db64ffb92084d8 from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

PPC doesn't appear to flush the cache on memunmap() but the
cache must be flushed since QMan does non coherent accesses
to this memory

Signed-off-by: Roy Pledge <roy.pledge@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/soc/fsl/qbman/qman_ccsr.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/soc/fsl/qbman/qman_ccsr.c b/drivers/soc/fsl/qbman/qman_ccsr.c
index bcf7f4e00241..0c521c9f3062 100644
--- a/drivers/soc/fsl/qbman/qman_ccsr.c
+++ b/drivers/soc/fsl/qbman/qman_ccsr.c
@@ -377,6 +377,16 @@ static int qm_set_memory(enum qm_memory memory, u64 ba, u32 size)
 		return -ENOMEM;
 	}
 	memset(ptr, 0, size);
+
+#ifdef CONFIG_PPC
+	/*
+	 * PPC doesn't appear to flush the cache on memunmap() but the
+	 * cache must be flushed since QMan does non coherent accesses
+	 * to this memory
+	 */
+	flush_dcache_range((unsigned long) ptr, (unsigned long) ptr+size);
+#endif
+
 	memunmap(ptr);
 
 	qm_ccsr_out(offset, upper_32_bits(ba));
-- 
2.17.1

